2013-09-23 19:15:18,899 - [INFO] - from play in main 
Listening for HTTP on /0.0.0.0:9000

2013-09-23 19:16:24,712 - [INFO] - from play in play-internal-execution-context-1 
datasource [jdbc:h2:mem:play] bound to JNDI as DefaultDS

2013-09-23 19:16:24,985 - [INFO] - from play in play-internal-execution-context-1 
database [default] connected at jdbc:h2:mem:play

2013-09-23 19:16:27,051 - [ERROR] - from application in New I/O worker #1 


! @6fl563kpg - Internal server error, for (GET) [/ddllab] ->

play.api.db.evolutions.InvalidDatabaseRevision: Database 'default' needs evolution![An SQL script need to be run on your database.]
	at play.api.db.evolutions.EvolutionsPlugin$$anonfun$onStart$1$$anonfun$apply$1.apply$mcV$sp(Evolutions.scala:462) ~[play-jdbc_2.10.jar:2.1.4]
	at play.api.db.evolutions.EvolutionsPlugin.withLock(Evolutions.scala:483) ~[play-jdbc_2.10.jar:2.1.4]
	at play.api.db.evolutions.EvolutionsPlugin$$anonfun$onStart$1.apply(Evolutions.scala:439) ~[play-jdbc_2.10.jar:2.1.4]
	at play.api.db.evolutions.EvolutionsPlugin$$anonfun$onStart$1.apply(Evolutions.scala:437) ~[play-jdbc_2.10.jar:2.1.4]
	at scala.collection.immutable.List.foreach(List.scala:309) ~[scala-library.jar:na]
	at play.api.db.evolutions.EvolutionsPlugin.onStart(Evolutions.scala:437) ~[play-jdbc_2.10.jar:2.1.4]
	at play.api.Play$$anonfun$start$1$$anonfun$apply$mcV$sp$1.apply(Play.scala:85) ~[play_2.10.jar:2.1.4]
	at play.api.Play$$anonfun$start$1$$anonfun$apply$mcV$sp$1.apply(Play.scala:85) ~[play_2.10.jar:2.1.4]
	at scala.collection.immutable.List.foreach(List.scala:309) ~[scala-library.jar:na]
	at play.api.Play$$anonfun$start$1.apply$mcV$sp(Play.scala:85) ~[play_2.10.jar:2.1.4]
	at play.api.Play$$anonfun$start$1.apply(Play.scala:85) ~[play_2.10.jar:2.1.4]
	at play.api.Play$$anonfun$start$1.apply(Play.scala:85) ~[play_2.10.jar:2.1.4]
	at play.utils.Threads$.withContextClassLoader(Threads.scala:18) ~[play_2.10.jar:2.1.4]
	at play.api.Play$.start(Play.scala:84) ~[play_2.10.jar:2.1.4]
	at play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(ApplicationProvider.scala:141) ~[play_2.10.jar:2.1.4]
	at play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(ApplicationProvider.scala:114) ~[play_2.10.jar:2.1.4]
	at scala.Option.map(Option.scala:145) ~[scala-library.jar:na]
	at play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1.apply(ApplicationProvider.scala:114) ~[play_2.10.jar:2.1.4]
	at play.core.ReloadableApplication$$anonfun$get$1$$anonfun$apply$1.apply(ApplicationProvider.scala:112) ~[play_2.10.jar:2.1.4]
	at scala.util.Either$RightProjection.flatMap(Either.scala:523) ~[scala-library.jar:na]
	at play.core.ReloadableApplication$$anonfun$get$1.apply(ApplicationProvider.scala:112) ~[play_2.10.jar:2.1.4]
	at play.core.ReloadableApplication$$anonfun$get$1.apply(ApplicationProvider.scala:104) ~[play_2.10.jar:2.1.4]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24) ~[scala-library.jar:na]
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24) ~[scala-library.jar:na]
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895) [na:1.6.0_51]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918) [na:1.6.0_51]
	at java.lang.Thread.run(Thread.java:680) [na:1.6.0_51]

2013-09-23 19:16:30,726 - [ERROR] - from play in New I/O worker #1 
Unknown data type: "("; SQL statement:
CREATE TABLE IF NOT EXISTS CLIENT
(
client_id INT NOT NULL AUTO_INCREMENT,
last VARCHAR(20) NOT NULL,
first VARCHAR(20) NOT NULL,
mi VARCHAR(1),
company VARCHAR(30),
street_address VARCHAR(50),
mailing_address VARCHAR(50),
city VARCHAR(30) NOT NULL,
state CHAR(2),
zip VARCHAR(10) NOT NULL,
office_phone CHAR(10) NOT NULL,
cell_phone CHAR(10),
sex CHAR(1),
fax CHAR(10),
reporting_email VARCHAR(30),
invoice_email VARCHAR(30),
phone_report_OK BIT NOT NULL,
email_report_OK BIT NOT NULL,
email_invoice_OK BIT NOT NULL,
bill_to_client INT,
PRIMARY KEY(client_id),
UNIQUE(first, last, mailing_address, city, state, zip),
INDEX (bill_to_client),
FOREIGN KEY (bill_to_client) REFERENCES CLIENT(client_id)
) [50004-168] [ERROR:50004, SQLSTATE:HY004]

2013-09-23 19:16:30,729 - [ERROR] - from application in New I/O worker #1 


! @6fl5640ba - Internal server error, for (GET) [/@evolutions/apply/default?redirect=http%3A%2F%2Flocalhost%3A9000%2Fddllab] ->

play.api.db.evolutions.InconsistentDatabase: Database 'default' is in inconsistent state![An evolution has not been applied properly. Please check the problem and resolve it manually before marking it as resolved.]
	at play.api.db.evolutions.Evolutions$.checkEvolutionsState(Evolutions.scala:177) ~[play-jdbc_2.10.jar:2.1.4]
	at play.api.db.evolutions.Evolutions$.applyScript(Evolutions.scala:275) ~[play-jdbc_2.10.jar:2.1.4]
	at play.api.db.evolutions.EvolutionsPlugin.handleWebCommand(Evolutions.scala:534) ~[play-jdbc_2.10.jar:2.1.4]
	at play.core.ReloadableApplication$$anonfun$handleWebCommand$12$$anonfun$apply$11.apply(ApplicationProvider.scala:282) ~[play_2.10.jar:2.1.4]
	at play.core.ReloadableApplication$$anonfun$handleWebCommand$12$$anonfun$apply$11.apply(ApplicationProvider.scala:281) ~[play_2.10.jar:2.1.4]
	at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:110) ~[scala-library.jar:na]
	at scala.collection.immutable.List.foldLeft(List.scala:78) ~[scala-library.jar:na]
	at play.core.ReloadableApplication$$anonfun$handleWebCommand$12.apply(ApplicationProvider.scala:281) ~[play_2.10.jar:2.1.4]
	at play.core.ReloadableApplication$$anonfun$handleWebCommand$12.apply(ApplicationProvider.scala:280) ~[play_2.10.jar:2.1.4]
	at scala.Option.flatMap(Option.scala:170) ~[scala-library.jar:na]
	at play.core.ReloadableApplication.handleWebCommand(ApplicationProvider.scala:280) ~[play_2.10.jar:2.1.4]
	at play.core.server.Server$$anonfun$getHandlerFor$1.apply(Server.scala:85) ~[play_2.10.jar:2.1.4]
	at play.core.server.Server$$anonfun$getHandlerFor$1.apply(Server.scala:85) ~[play_2.10.jar:2.1.4]
	at scala.util.control.Exception$Catch$$anonfun$either$1.apply(Exception.scala:124) ~[scala-library.jar:na]
	at scala.util.control.Exception$Catch$$anonfun$either$1.apply(Exception.scala:124) ~[scala-library.jar:na]
	at scala.util.control.Exception$Catch.apply(Exception.scala:102) ~[scala-library.jar:na]
	at scala.util.control.Exception$Catch.either(Exception.scala:124) ~[scala-library.jar:na]
	at play.core.server.Server$class.getHandlerFor(Server.scala:85) ~[play_2.10.jar:2.1.4]
	at play.core.server.NettyServer.getHandlerFor(NettyServer.scala:35) [play_2.10.jar:2.1.4]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$7.apply(PlayDefaultUpstreamHandler.scala:112) [play_2.10.jar:2.1.4]
	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$7.apply(PlayDefaultUpstreamHandler.scala:112) [play_2.10.jar:2.1.4]
	at scala.util.Either.fold(Either.scala:98) [scala-library.jar:na]
	at play.core.server.netty.PlayDefaultUpstreamHandler.messageReceived(PlayDefaultUpstreamHandler.scala:106) [play_2.10.jar:2.1.4]
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70) [netty.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:560) [netty.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:787) [netty.jar:na]
	at com.typesafe.netty.http.pipelining.HttpPipeliningHandler.messageReceived(HttpPipeliningHandler.java:64) [netty-http-pipelining.jar:na]
	at org.jboss.netty.channel.SimpleChannelHandler.handleUpstream(SimpleChannelHandler.java:88) [netty.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:560) [netty.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:787) [netty.jar:na]
	at org.jboss.netty.handler.codec.http.HttpContentDecoder.messageReceived(HttpContentDecoder.java:108) [netty.jar:na]
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70) [netty.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:560) [netty.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:787) [netty.jar:na]
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:296) [netty.jar:na]
	at org.jboss.netty.handler.codec.frame.FrameDecoder.unfoldAndFireMessageReceived(FrameDecoder.java:459) [netty.jar:na]
	at org.jboss.netty.handler.codec.replay.ReplayingDecoder.callDecode(ReplayingDecoder.java:536) [netty.jar:na]
	at org.jboss.netty.handler.codec.replay.ReplayingDecoder.messageReceived(ReplayingDecoder.java:435) [netty.jar:na]
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70) [netty.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:560) [netty.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:555) [netty.jar:na]
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:268) [netty.jar:na]
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:255) [netty.jar:na]
	at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:88) [netty.jar:na]
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:107) [netty.jar:na]
	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:312) [netty.jar:na]
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:88) [netty.jar:na]
	at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178) [netty.jar:na]
	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) [netty.jar:na]
	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) [netty.jar:na]
	at java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:895) [na:1.6.0_51]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:918) [na:1.6.0_51]
	at java.lang.Thread.run(Thread.java:680) [na:1.6.0_51]

2013-09-23 19:16:37,449 - [INFO] - from play in play-internal-execution-context-2 
datasource [jdbc:h2:mem:play] bound to JNDI as DefaultDS

2013-09-23 19:16:37,452 - [INFO] - from play in play-internal-execution-context-2 
database [default] connected at jdbc:h2:mem:play

2013-09-23 19:16:37,816 - [INFO] - from play in play-internal-execution-context-2 
Application started (Dev)

